src/particles/plasma/PlasmaParticleContainer.cpp

```diff
@@ -448,6 +452,203 @@ IonizationModule (const int lev,
+                                         (PhysConstSI::ep0 * PhysConstSI::m_e) );
+        const amrex::Real E0 = Hipace::m_normalized_units ?
+                               wp * PhysConstSI::m_e * PhysConstSI::c / PhysConstSI::q_e : 1;
+	const amrex::Real lambda0 = laser.GetLambda0();
+        const amrex::Real omega0 = 2.0 * MathConst::pi * phys_const.c / lambda0;
+	
+        int * const ion_lev = soa_ion.GetIntData(PlasmaIdx::ion_lev).data();
+        const amrex::Real * const x_prev = soa_ion.GetRealData(PlasmaIdx::x_prev).data();
+        const amrex::Real * const y_prev = soa_ion.GetRealData(PlasmaIdx::y_prev).data();
+        const amrex::Real * const uxp = soa_ion.GetRealData(PlasmaIdx::ux_half_step).data();
+        const amrex::Real * const uyp = soa_ion.GetRealData(PlasmaIdx::uy_half_step).data();
+        const amrex::Real * const psip =soa_ion.GetRealData(PlasmaIdx::psi_half_step).data();
+        const auto * idcpup = soa_ion.GetIdCPUData().data();
+
+        // Make Ion Mask and load ADK prefactors
+        // Ion Mask is necessary to only resize electron particle tile once
+        amrex::Gpu::DeviceVector<uint8_t> ion_mask(ptile_ion.numParticles(), 0);
+        uint8_t* AMREX_RESTRICT p_ion_mask = ion_mask.data();
+        amrex::Gpu::DeviceScalar<uint32_t> num_new_electrons(0);
+        uint32_t* AMREX_RESTRICT p_num_new_electrons = num_new_electrons.dataPtr();
+        amrex::Real* AMREX_RESTRICT adk_prefactor = m_adk_prefactor.data();
+        amrex::Real* AMREX_RESTRICT adk_exp_prefactor = m_adk_exp_prefactor.data();
+        amrex::Real* AMREX_RESTRICT adk_power = m_adk_power.data();
+        amrex::Real* AMREX_RESTRICT laser_adk_prefactor = m_laser_adk_prefactor.data();
+
+        long num_ions = ptile_ion.numParticles();
+
+        amrex::AnyCTO(
+            amrex::TypeList<
+                amrex::CompileTimeOptions<0, 1, 2, 3>
+            >{}, {
+                Hipace::m_depos_order_xy
+            },
+            [&] (auto cto_func) {
+                amrex::ParallelForRNG(num_ions, cto_func);
+            },
+            [=] AMREX_GPU_DEVICE (long ip, const amrex::RandomEngine& engine,
+                                  auto depos_order_xy) {
+
+            // avoid temp slice
+            const amrex::Real xp = x_prev[ip];
+            const amrex::Real yp = y_prev[ip];
+
+            if (amrex::ConstParticleIDWrapper(idcpup[ip]) < 0 ||
+                !laser_bounds.contains(xp, yp)) return;
+
+            Complex A = 0;
+            Complex A_dx = 0;
+            Complex A_dzeta = 0;
+
+            // TODO convert units of A
+            // this is likely incorrect
+            doLaserGatherShapeN<depos_order_xy>(xp, yp, A, A_dx, A_dzeta, laser_arr,
+                dx_inv, dy_inv, dzeta_inv, x_pos_offset, y_pos_offset);
+
+            const Complex Et = I * A * omega0 + A_dzeta * phys_const.c; //transverse component
+            const Complex El = - A_dx * phys_const.c; //longitudinal component
```

Comment:
For simple tests, these could be replaced as follows
```suggestion
            const Complex Et = I * A * omega0; //transverse component
            const Complex El = 0; //longitudinal component
```
i.e. only keeping the leading order to remove potential sources of error.